test/Chainweb/Test/Mempool.hs

```diff
@@ -219,6 +220,35 @@ propBadlist (txs, badTxs) _ mempool = runExceptT $ do
     insert v = mempoolInsert mempool CheckedInsert $ V.fromList v
     lookup = mempoolLookup mempool . V.fromList . map hash
 
+propAddToBadList
+  :: MockTx
+  -> InsertCheck
+  -> MempoolBackend MockTx
+  -> IO (Either String ())
+propAddToBadList tx _ mempool = runExceptT $ do
+    liftIO (insert [tx])
+    liftIO (lookup [tx]) >>= V.mapM_ lookupIsPending
+
+    block <- getBlock
+    when (block /= [tx]) $ fail "expected to get our tx back"
+
+    liftIO $ mempoolAddToBadList mempool $ hash tx
+    block' <- getBlock
+    when (block' /= []) $ fail "expected to get an empty block"
+    liftIO (lookup [tx]) >>= V.mapM_ lookupIsMissing
+    liftIO (insert [tx])
+    liftIO (lookup [tx]) >>= V.mapM_ lookupIsMissing
+
```

Comment:
Would it be possible to extend the test scenario to also check that the bad-list gets cleaned up and isn't growing forward?

Reply:
Maybe later, you're right that we aren't testing badmap pruning currently. Testing it will probably require deeper changes to some of the code (in particular to pass the clock around as an IO action for testing) so I'd like to do it later.

To answer your other question in case you're worried about this though, we do have telemetry on the size of the badmap so we'll be able to see if it is growing indefinitely.